src/main/kotlin/com/toasttab/pulseman/state/ReceiveMessage.kt

```diff
@@ -53,17 +55,20 @@ class ReceiveMessage(
     private var consumer: Consumer<ByteArray>? = null
 
     private val stateVertical = ScrollState(0)
+    private var subscribeFuture: CompletableFuture<Consumer<ByteArray>>? = null
```

Comment:
Make this global so it can be closed mid connection attempt, it was hanging previously